src/simd/distances_avx512.cc

```diff
@@ -159,19 +159,17 @@ FAISS_PRAGMA_IMPRECISE_FUNCTION_END
 float
 fp16_vec_inner_product_avx512(const knowhere::fp16* x, const knowhere::fp16* y, size_t d) {
     __m512 m512_res = _mm512_setzero_ps();
-    __m512 m512_res_0 = _mm512_setzero_ps();
```

Comment:
`res_0` is used for increasing the instruction level parallelism in the following loop.
Please confirm with godbolt.org, or let me know if I need to check this.

Reply:
There is a slight difference between fp16_vec_inner_product_avx512 and fp16_vec_inner_product_avx512_batch_4: 
fp16_vec_inner_product_avx512 : sum of (round(a*b + m512_res)) +  sum of (round(a*b + m512_res_0)) 
fp16_vec_inner_product_avx512_batch_4: sum of (round(a*b + m512_res))

Reply:
presicion loss may caused by fmadd: round(a*b + c)